tools/last_user_activity.py

```diff
@@ -0,0 +1,340 @@
+
+        # Get all members from all orgs
+        all_members = {}
+        for org in orgs:
+            members = await get_org_members(session, org, debug)
+            for member in members:
+                if member["login"] not in all_members:
+                    all_members[member["login"]] = []
+                all_members[member["login"]].append(org)
+
+        # Get activity for each user
+        tasks = []
+        for username in all_members:
+            task = get_user_activity(session, username, debug)
+            tasks.append((username, task))
+
+        results = await asyncio.gather(*(task for _, task in tasks))
+
+        # Print results sorted by last activity
+        user_activities = []
+        for (username, _), last_activity in zip(tasks, results):
+            if last_activity is not None:
+                assert isinstance(last_activity, datetime), last_activity
+            user_activities.append(
+                (
+                    username,
+                    last_activity if last_activity is not None else None,
+                    all_members[username],
+                )
+            )
+        for org in orgs:
+            print(f"[bold]{org}[/bold]")
+            n_active = 0
+            n_inactive = 0
+            for username, last_activity, user_orgs in sorted(
+                user_activities,
+                key=lambda x: (x[1], x[0])
+                if x[1] is not None
+                else (datetime.fromtimestamp(0).replace(tzinfo=timezone.utc), x[0]),
+                reverse=True,
+            ):
+                if org not in user_orgs:
+                    continue
+                if last_activity is not None and last_activity > (datetime.now().replace(tzinfo=timezone.utc) - timedelta(days=timelimit_days)):
+                    n_active += 1
+                    continue
+                n_inactive += 1
+                last_activity_ago = (
+                    humanize.naturaltime(datetime.now(last_activity.tzinfo) - last_activity)
+                    if last_activity
+                    else "[red]never[/red]"
+                )
+                orgs_str = ", ".join(user_orgs)
+                print(
+                    f"    {username:<20}: Last activity {last_activity_ago}"
+                )
+            print(f"    Found [red]{n_inactive} inactive[/red] and [green]{n_active} active[/green] users in {org} with last activity more recent than {timelimit_days} days.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="GitHub Organization Activity Tracker")
+    parser.add_argument(
+        "--clear-cache", action="store_true", help="Clear the cache before running"
+    )
+    parser.add_argument("--debug", action="store_true", help="Show debug information")
+
+
+    parser.add_argument(
+        "--timelimit-days",
+        type=int,
+        default=0,
+        help="Time limit in days for the last activity (default: 30)",
+    )
```

Comment:
The default value of `0` here results in all users being listed as inactive by default.

We should probably change this to 365, per discussion at the Security call.

```suggestion
    parser.add_argument(
        "--timelimit-days",
        type=int,
        default=365,
        help="Maximum number of days since last activity before an account is marked as inactive. (default: 365)",
    )
```